### ISSUE TITLE
Intl.Segments.prototype.containing should include undefined in return value

### ISSUE THREAD

**Issue by Jamesernator** [microsoft/TypeScript#61235]
### ⚙ Compilation target

ESNext

### ⚙ Library

ESNext

### Missing / Incorrect Definition

The interface `Intl.Segments` lacks an `undefined` in the return type for `containing`:

```ts
interface Segments {
    // Should be SegmentData | undefined
    containing(codeUnitIndex?: number): SegmentData;
}
```

### Sample Code

```TypeScript
const segmenter = new Intl.Segmenter();

const segments = segmenter.segment("Hello world");
// @ts-expect-error This should report an error
const s: Intl.SegmentData = segments.containing(40);
// This should be fine
const s: Intl.SegmentData | undefined = segments.containing(40);
```

### Documentation Link

MDN: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Intl/Segmenter/segment/Segments/containing#return_value
